server/src/api/v1.py

```diff
@@ -199,8 +229,7 @@ def job_builder(data: dict, auth_token: str):
     allowed = check_token_permissions(
```

Comment:
Here, you are returning `allowed` but if that is false you have lost the information about _why_ it is false, i.e. exactly what the permission problem is. So right underneath, you are aborting with an error message that has to do with priorities, even though the rejection might have different reasons, e.g. max reservation time. So perhaps you can move the (different) `abort`s inside the `check_token_permissions` function and display an appropriate message depending on the permission issue. Returning successfully from `check_token_permissions` will mean there are no permission issues.